subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.c

```diff
@@ -143,8 +150,11 @@ static int prepare_cb(struct lll_prepare_param *prepare_param)
 	}
 
 	radio_reset();
-	/* TODO: other Tx Power settings */
+#if defined(CONFIG_BT_TX_PWR_DYNAMIC_CTRL)
+	radio_tx_power_set((u32_t)((s32_t)lll->tx_pwr_lvl));
```

Comment:
I think it can be a bit improved by casting directly to `u32_t` as the `s8_t` to `s32_t` extension should happen automatically. The reason I use `s8_t` is very simple. The memory footprint scales with the no. of connections allowed and each connection would then have 3 extra bytes which are unused and unnecessary.

On the other hand, on the radio low-level side the inputs modify directly the 32bit registers, hence the `u32_t` type for the radio Tx power set interface.

Reply:
If we use s8_t as the type, then it should be that for all functions accepting this parameter. The cast to u32_t for the HW mmio can be done inside that particular function.
My 2 cents :)

Reply:
@cvinayak would you agree for the HAL setter so I could rework this ?

Reply:
yes. Have it as separate commit.

Reply:
Done in 6716117a10